addon-resizer/nanny/kubernetes_client_test.go

```diff
@@ -100,3 +103,185 @@ func TestMergeResources(t *testing.T) {
 		})
 	}
 }
+
+func expectErrorOrCount(t *testing.T, expectErr error, expectCount uint64, gotErr error, gotCount uint64) {
+	if expectErr == nil {
+		if gotErr != nil {
+			t.Errorf("expected no error, got %v", gotErr)
+		}
+		if gotCount != expectCount {
+			t.Errorf("expected node count %v, got %v", expectCount, gotCount)
+		}
+		return
+	}
+	if gotErr == nil {
+		t.Errorf("expected error %v, got nil", expectErr)
+		return
+	}
+	if gotErr.Error() != expectErr.Error() {
+		t.Errorf("expected error %v, got %v", expectErr, gotErr)
+	}
+}
+
+func getMetric(labelName, labelValue string, value float64) *dto.Metric {
+	return &dto.Metric{
+		Label: []*dto.LabelPair{
+			{
+				Name:  &labelName,
+				Value: &labelValue,
+			},
+		},
+		Gauge: &dto.Gauge{
+			Value: &value,
+		},
+	}
+}
+
+func TestExtractMetricValueForNodeCount(t *testing.T) {
```

Comment:
Should we test:

Multiple labels on a metric?
Multiple metrics in metricFamily?

Reply:
Done